toolkit/tools/pkg/imagecustomizerlib/telemetry.go

```diff
@@ -0,0 +1,78 @@
+package imagecustomizerlib
+
+import (
+	"context"
+	"net"
+	"runtime"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+	"go.opentelemetry.io/otel/trace/noop"
+)
+
+var (
+	Tracer trace.Tracer
+
+	TracerName = "imagecustomizer"
+
+	TelemetryServerAddress = "localhost:4317"
+
+	shutdownFn func(ctx context.Context) error
+)
+
+func InitTelemetry(disableTelemetry bool) error {
+	if disableTelemetry || !checkServerListening() {
+		// Use noop tracer when telemetry is disabled
+		logger.Log.Trace("Disabled telemetry collection")
+		Tracer = noop.NewTracerProvider().Tracer(TracerName)
+		shutdownFn = nil
+		return nil
+	}
+
+	exporter, err := otlptracegrpc.New(context.Background(),
+		otlptracegrpc.WithInsecure(),
+		otlptracegrpc.WithEndpoint(TelemetryServerAddress),
+	)
+	if err != nil {
+		logger.Log.Tracef("Failed to create OTLP exporter: %v", err)
+		return err
+	}
+
+	distro, version := getDistroAndVersion()
+
+	tp := sdktrace.NewTracerProvider(
+		sdktrace.WithBatcher(exporter),
+		sdktrace.WithResource(resource.NewSchemaless(
+			attribute.String("host.architecture", runtime.GOARCH),
+			attribute.String("host.os", distro),
+			attribute.String("host.os.version", version),
+			attribute.String("imagecustomizer.version", ToolVersion),
+		)),
+	)
+	otel.SetTracerProvider(tp)
+	Tracer = tp.Tracer(TracerName)
```

Comment:
`otel.GetTracerProvider()` will return a no-op provider if `otel.SetTracerProvider` hasn't been called. In addition, `tp.Tracer` will return the same tracer if passed the same name.

So, in library code, you can just call `otel.GetTracerProvider().Tracer(name)`. There is no need to store the tracer in a global variable.

Also, looking at [the docs](https://pkg.go.dev/go.opentelemetry.io/otel/trace#TracerProvider), I think the tracer's name should match the name of the Go package, not the name of the executable.